crates/syntax/src/ast/edit_in_place.rs

```diff
@@ -236,6 +236,17 @@ impl ast::GenericParamList {
         }
     }
 
+    /// Removes the existing generic param
+    pub fn remove_generic_param(&self, generic_param: ast::GenericParam) {
+        if let Some(previous) = generic_param.syntax().prev_sibling() {
+            if let Some(next_token) = previous.next_sibling_or_token() {
+                ted::remove_all(next_token..=generic_param.syntax().clone().into());
+            }
+        } else {
+            ted::remove(generic_param.syntax());
+        }
```

Comment:
This should also have a check for a following sibling, if there is not previous one to be found

Reply:
Good observation, have updated the code accordingly and added a test to check the behaviour is expected.